catalog-rest-service/src/main/java/org/openmetadata/catalog/jdbi3/EntityRepository.java

```diff
@@ -451,19 +451,22 @@ public final T createInternal(T entity) throws IOException, ParseException {
   }
 
   @Transaction
-  public final DeleteResponse<T> delete(String updatedBy, String id) throws IOException {
+  public final DeleteResponse<T> delete(String updatedBy, String id) throws IOException, ParseException {
     return delete(updatedBy, id, false);
   }
 
   @Transaction
-  public final DeleteResponse<T> delete(String updatedBy, String id, boolean recursive) throws IOException {
+  public final DeleteResponse<T> delete(String updatedBy, String id, boolean recursive)
+      throws IOException, ParseException {
     // Validate entity
     String json = dao.findJsonById(id, Include.NON_DELETED);
     if (json == null) {
       throw EntityNotFoundException.byMessage(CatalogExceptionMessage.entityNotFound(entityType, id));
     }
 
     T original = JsonUtils.readValue(json, entityClass);
+    setFields(original, putFields);
```

Comment:
This is the only change, the rest are propagating the `ParseException` in the function signature